src/Samples/DesktopServer/Program.cs

```diff
@@ -1,25 +1,42 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.ServiceModel;
+using CoreWCF.Samples.StandardCommon;
 
 namespace DesktopServer
 {
     class Program
     {
-        static void Main(string[] args)
+        static void Main()
         {
-            var contract = typeof(Contract.IEchoService);
-            var host = new ServiceHost(typeof(EchoService),
-                new Uri("net.tcp://localhost:8089/"),
-                new Uri("http://localhost:8088/"),
-                new Uri("https://localhost:8443/"));
+            Settings settings = new Settings().SetDetaults();
 
-            host.AddServiceEndpoint(contract, new NetTcpBinding(), "/nettcp");
-            host.AddServiceEndpoint(contract, new BasicHttpBinding(BasicHttpSecurityMode.None), "/basichttp");
-            host.AddServiceEndpoint(contract, new BasicHttpsBinding(BasicHttpsSecurityMode.Transport), "/basichttp");
-            host.AddServiceEndpoint(contract, new WSHttpBinding(SecurityMode.None), "/wsHttp.svc");
-            host.AddServiceEndpoint(contract, new WSHttpBinding(SecurityMode.Transport), "/wsHttp.svc");
+            string hostname = "localhost"; //"localtest"
+
+            IEnumerable<string> baseAddressList = new[] {
+                "net.tcp://localhost:8089/",
+                "http://localhost:8088/",
+                "https://localhost:8443/" };
+            baseAddressList = baseAddressList.Select(i => i.Replace("//localhost:", $"//{hostname}:"));
+            Uri[] baseUriList = baseAddressList.Select(a => new Uri(a)).ToArray();
```

Comment:
This replacing of the hostname isn't needed as you added an optional parameter to SetDefaults to do this for you.